spec/forms/steps/application/declaration_form_spec.rb

```diff
@@ -23,18 +23,53 @@
     end
 
     context 'validations' do
+      before do
+        allow(c100_application).to receive(:valid?)
+      end
+
       it { should validate_presence_of(:declaration_signee) }
       it { should validate_presence_of(:declaration_signee_capacity, :inclusion) }
     end
 
-    context 'when form is valid' do
-      it 'saves the record' do
-        expect(c100_application).to receive(:update).with(
-          declaration_signee: 'Full Name',
-          declaration_signee_capacity: 'applicant',
-        ).and_return(true)
+    context 'when application fulfilment is successful' do
+      before do
+        allow(c100_application).to receive(:valid?).with(:completion).and_return(true)
+      end
+
+      context 'and form is valid' do
+        it 'saves the record' do
+          expect(c100_application).to receive(:update).with(
+            declaration_signee: 'Full Name',
```

Comment:
it seems to me that you can use the values of the `let`'s on line 11 and 12 to fill the values here.